spec/controllers/relationships_controller_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'spec_helper'
+
+describe RelationshipsController, type: :controller do
+  context "User is signed in" do
+    describe "follow and unfollow" do
+      before do
+        @user1 = FactoryGirl.create(:user)
+        sign_in(@user1)
+        @user2 = FactoryGirl.create(:user, email: "t@t.com", username: "tester")
+      end
+
+      it "allows user to follow other users" do
+        expect {post :follow, :id => @user2.username}.to change {@user2.followers.count}.from(0).to(1)
```

Comment:
We should also test how the controller responds to JS requests vs HTML requests. We expect a redirect in one case, and a 200 in the other.